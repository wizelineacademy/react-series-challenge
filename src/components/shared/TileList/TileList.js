import React from "react";

import Tile from "../Tile";
import { Loader } from "../../shared";
import { TilesWrapperStyled } from "./TileList.styled";

const TileList = ({ gifs, loading }) => {
  if (loading) return <Loader />;

  return (
    <TilesWrapperStyled>
      {gifs.map(gif => <Tile key={gif.id} gif={gif} />)}
    </TilesWrapperStyled>
  );
};

export default TileList;
