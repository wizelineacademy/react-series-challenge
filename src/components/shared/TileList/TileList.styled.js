import styled from "styled-components";

const TilesWrapperStyled = styled.div`
  display: flex;
  width: 100vw;
  flex-wrap: wrap;
  justify-content: center;
`;

export { TilesWrapperStyled };
