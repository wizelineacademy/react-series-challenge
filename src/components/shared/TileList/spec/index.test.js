import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import configureStore from "redux-mock-store";
import "jest-styled-components";
import TileList from "../";

const mockStore = configureStore();

const shape = {
  loading: false,
  gifs: [
    {
      id: 1,
      images: {
        original: { width: 400, height: 400, url: "http://original" },
        fixed_width: { width: 300, height: 300, url: "http://fixed" }
      }
    },
    {
      id: 2,
      images: {
        original: { width: 400, height: 400, url: "http://original" },
        fixed_width: { width: 300, height: 300, url: "http://fixed" }
      }
    }
  ]
};

const initialState = {
  favsReducer: { favs: [] },
  imageLoaderReducer: { loaded: {} }
};

describe("<TileList />", () => {
  it("renders properly", () => {
    const init = Object.assign(initialState);
    const store = mockStore(init);
    const props = Object.assign(shape);
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <TileList {...props} />
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the loader", () => {
    const props = {
      gifs: {},
      loading: true
    };
    ReactDOM.createPortal = node => node;
    const component = renderer.create(<TileList {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
