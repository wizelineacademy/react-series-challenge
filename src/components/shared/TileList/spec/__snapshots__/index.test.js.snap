// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileList /> renders properly 1`] = `
.c1 {
  width: 300px;
  height: 300px;
  position: relative;
  margin: 5px;
  -webkit-transition: 0.1s ease-in-out -webkit-transform;
  -webkit-transition: 0.1s ease-in-out transform;
  transition: 0.1s ease-in-out transform;
  border-radius: 2px;
  background: linear-gradient(to right,#eeeeee 8%,#dddddd 18%,#eeeeee 33%);
  -webkit-animation: 1.5s eNVqyK linear infinite;
  animation: 1.5s eNVqyK linear infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  background-size: 800px 104px;
  position: relative;
}

.c1:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100vw;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  <a
    href="/details/1"
    onClick={[Function]}
  >
    <div
      className="c1"
      height={300}
      width={300}
    />
  </a>
  <a
    href="/details/2"
    onClick={[Function]}
  >
    <div
      className="c1"
      height={300}
      width={300}
    />
  </a>
</div>
`;

exports[`<TileList /> renders the loader 1`] = `
.c0 {
  position: absolute;
  top: 0;
  right: 18px;
  width: 25px;
  height: 25px;
  border: 3px solid white;
  border-left-color: deepskyblue;
  border-radius: 50%;
  -webkit-animation: 1s linear iVXCSc infinite;
  animation: 1s linear iVXCSc infinite;
}

<div
  className="c0"
/>
`;
